### PR TITLE
Multistore - no storeCode in URL

### PR DESCRIPTION
**Pull request by jpetar**
This change would handle the case in which your Multistore doesn't use the storeCode in the URL for the main store.

### CODE REVIEW

File: router/beforeEach.ts
Comment (ResuBaka) [inline router/beforeEach.ts]:
```typescript
} else if (storeCode in stores && to.name === 'checkout'  && stores[storeCode].cmsUrl !== undefined) {
  window.location.replace(stores[storeCode].cmsUrl + '/vue/cart/sync/token/' + userToken + '/cart/' + cartToken)
}
```

Isn't that the right code we would need here ?

**Review by ResuBaka — COMMENTED** [review]
To get the right storeCode we should use `import {currentStoreView} from '@vue-storefront/core/lib/multistore'` and from that `currentStoreView().storeCode` to get the storeCode from the current store.

File: router/beforeEach.ts
Comment (jpetar) [reply router/beforeEach.ts]: In my case your if condition still wouldn't work because of the `to.name === 'checkout'` but I guess you could use `currentStoreView().storeCode`.

Comment (ResuBaka) [reply router/beforeEach.ts]: That i don't really understand? 

You are doing the `to.name === 'checkout'` too or are you meaning something else that would not work for you?

Comment (jpetar) [reply router/beforeEach.ts]: Sorry, you're right. 

I meant the `if (storeCode in stores` part. This doesn't work because of  `const storeCode = storeCodeFromRoute(to)`.
In my case there is no storeCode in the url of the main store.

Comment (ResuBaka) [reply router/beforeEach.ts]: That is why the `const storeCode = storeCodeFromRoute(to)` should be replaced with `const storeCode = currentStoreView().storeCode` then you have your storeCode.

Comment (jpetar) [reply router/beforeEach.ts]: I see it now. I'm gonna try to commit this in my PR branch.

Comment (jpetar) [reply router/beforeEach.ts]: I've commited your solution to the branch.

**Review by ResuBaka — APPROVED** [review]

File: router/beforeEach.ts
Comment (ResuBaka) [reply router/beforeEach.ts]: Tanks for the fix.

Now we only need to wait until one finds the time to merge this.

**Comment by Tjitse-E** [comment]
@ResuBaka does this fix your issue #13?

**Comment by ResuBaka** [comment]
Yes this does fix the issue #13.

**Pull request merged**
